subprojects/ide-native/src/main/java/org/gradle/ide/xcode/plugins/XcodePlugin.java

```diff
@@ -247,37 +247,41 @@ private void configureXcodeForSwift(final Project project) {
             public void execute(Project project) {
                 // TODO: Assumes there's a single 'main' Swift component
                 ProductionSwiftComponent component = project.getComponents().withType(ProductionSwiftComponent.class).getByName("main");
-                FileCollection sources = component.getSwiftSource();
-                xcodeProject.getGroups().getSources().from(sources);
 
-                // TODO - should use the _install_ task for an executable
-                String targetName = component.getModule().get();
-                final XcodeTarget target = newTarget(targetName, component.getModule().get(), toGradleCommand(project), getBridgeTaskPath(project), sources);
-                component.getBinaries().whenElementFinalized(new Action<SwiftBinary>() {
-                    @Override
-                    public void execute(SwiftBinary swiftBinary) {
-                        if (swiftBinary instanceof SwiftExecutable && swiftBinary.isDebuggable() && !swiftBinary.isOptimized()) {
-                            target.setDebug(((SwiftExecutable) swiftBinary).getDebuggerExecutableFile(), PBXTarget.ProductType.TOOL);
-                        } else if (swiftBinary instanceof SwiftExecutable && swiftBinary.isDebuggable() && swiftBinary.isOptimized()) {
-                            target.setRelease(((SwiftExecutable) swiftBinary).getDebuggerExecutableFile(), PBXTarget.ProductType.TOOL);
-                        } else if (swiftBinary instanceof SwiftSharedLibrary && swiftBinary.isDebuggable() && !swiftBinary.isOptimized()) {
-                            target.setDebug(((SwiftSharedLibrary) swiftBinary).getRuntimeFile(), PBXTarget.ProductType.DYNAMIC_LIBRARY);
-                        } else if (swiftBinary instanceof SwiftSharedLibrary && swiftBinary.isDebuggable() && swiftBinary.isOptimized()) {
-                            target.setRelease(((SwiftSharedLibrary) swiftBinary).getRuntimeFile(), PBXTarget.ProductType.DYNAMIC_LIBRARY);
-                        } else if (swiftBinary instanceof SwiftStaticLibrary && swiftBinary.isDebuggable() && !swiftBinary.isOptimized()) {
-                            target.setDebug(((SwiftStaticLibrary) swiftBinary).getLinkFile(), PBXTarget.ProductType.STATIC_LIBRARY);
-                        } else if (swiftBinary instanceof SwiftStaticLibrary && swiftBinary.isDebuggable() && swiftBinary.isOptimized()) {
-                            target.setRelease(((SwiftStaticLibrary) swiftBinary).getLinkFile(), PBXTarget.ProductType.STATIC_LIBRARY);
+                // Don't configure anything if this component does not have a buildable development binary
```

Comment:
This still looks a little weird.  We're calling `isPresent` before `binaries.whenElementFinalized`, which sort of defeats the purpose of `whenElementFinalized` because the development binary has to be ready earlier.  And we explicitly reference the development binary (`getDevelopmentBinary().get()`) after the `whenElementFinalized` too.

Can this be written so that we rely on `whenElementFinalized` only and get rid of the afterEvaluate? The same goes for the C++ binary handling further down too.

Reply:
Good point.  I've moved this check inside the `whenElementFinalized` so that we're not doing the weird `isPresent()` check beforehand.  It still relies on the development binary being set before we realize the binaries container, so its still not as tight as we would like, but it's probably a little better than before.

I think we still need the `afterEvaluate` block because there is other information we use such as `SwiftBinary.module` and `CppBinary.baseName` that are mutable and could potentially change.